aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1391,12 +1370,13 @@ impl AptosVM {
         match &authenticators {
             Ok(authenticators) => {
                 for (_, sig) in authenticators {
-                    if !self.features.is_zkid_enabled()
+                    if !self.features().is_zkid_enabled()
                         && matches!(sig.sig, ZkpOrOpenIdSig::Groth16Zkp { .. })
                     {
                         return Err(VMStatus::error(StatusCode::FEATURE_UNDER_GATING, None));
                     }
-                    if (!self.features.is_zkid_enabled() || !self.features.is_zkid_zkless_enabled())
+                    if (!self.features().is_zkid_enabled()
```

Comment:
Oh, that might be a bug😅 rebase went wrong it seems

Reply:
I think `!self.features().is_zkid_enabled()` is a bit better because it aligns with `!self.features().is_zkid_enabled()` in the if above, but tbh the loop there looks a bit ugly, I will restructure it a bit then

Reply:
actually I think there is a logical problem.... @heliuchuan here we traverse the whole vector, but if you look at `validate_zkid_authenticators` code, it checks if # authenticators is 0 and < max allowed. The checks must be before we gate?